### PR TITLE
Decreasing required memory

### PR DESCRIPTION
**Pull request by schubi2**
Replacing old Pkg.ResolvableProperties()and Pkg.ResolvabeDependencies() calls by Y2Packager::Resolvable.any? and Y2Packager::Resolvable.find in order to save memory.

### CODE REVIEW

File: src/modules/ImageInstallation.rb [hunk 1]
```diff
@@ -1273,28 +1273,28 @@ def ProceedWithSelected(one_object, one_type)
...
-          resolvable_properties
+          resolvable_properties.map(&:name)
         )
       end
 
       # Leave only already installed (and matching the same architecture)
       resolvable_properties = Builtins.filter(resolvable_properties) do |one_resolvable|
-        Ops.get_symbol(one_resolvable, "status", :unknown) == :installed &&
-          Ops.get_string(one_resolvable, "arch", "") == arch
+        one_resolvable.status == :installed &&
```
Comment (lslezak) [inline src/modules/ImageInstallation.rb]: I'd move this filtering directly to the `Resolvable.find` arguments above.

File: src/modules/ImageInstallation.rb [hunk 2]
```diff
@@ -1273,28 +1273,28 @@ def ProceedWithSelected(one_object, one_type)
...
+          one_resolvable.arch == arch
       end
 
       if ThisIsADebugMode()
-        Builtins.y2milestone("Resolvables installed: %1", resolvable_properties)
+        Builtins.y2milestone("Resolvables installed: %1", resolvable_properties.map(&:name))
```
Comment (lslezak) [inline src/modules/ImageInstallation.rb]: I'd completely remove `ThisIsADebugMode()` and use `log.debug` here.

File: src/modules/ImageInstallation.rb [hunk 3]
```diff
@@ -1349,43 +1349,41 @@ def RestoreAllChanges
       @generic_set_progress.call(id, 0) if !@generic_set_progress.nil?
 
       Builtins.foreach(@all_supported_types) do |one_type|
```
Comment (lslezak) [inline src/modules/ImageInstallation.rb]: NP I was thinking about a complete refactoring of this but as we do not install images anymore we could probably drop it. And that's out of scope here...

File: src/lib/installation/clients/inst_addon_update_sources.rb [hunk 4]
```diff
@@ -152,41 +155,32 @@ def KnownUrls
...
-      products = Pkg.ResolvableProperties("", :product, "")
+      products = Y2Packager::Resolvable.find(kind: :product)
 
       Builtins.foreach(products) do |p|
-        Builtins.foreach(Ops.get_list(p, "update_urls", [])) do |u|
+        Builtins.foreach(p.update_urls) do |u|
           # bnc #542792
           # Repository name must be generated from product details
+          p_name =
+            [p.display_name, p.name, p.summary, _("Unknown Product")].reject(&:empty?).first
```
Comment (imobachgs) [inline src/lib/installation/clients/inst_addon_update_sources.rb]: I am starting to think that we should extend [Y2Packager::Resolvable](https://github.com/yast/yast-packager/blob/a58eb3ccda478a5908fb7d1e3bf4c65b64ce1bd6/src/lib/y2packager/resolvable.rb) with a `#human_name` or something similar.

File: src/lib/installation/clients/inst_addon_update_sources.rb [hunk 5]
```diff
@@ -152,41 +155,32 @@ def KnownUrls
...
-                  Ops.get_locale(p, "summary", _("Unknown Product"))
-                )
-              ),
-              Ops.get_string(p, "version", "")
+              p_name,
+              p.version
             )
           )
           # alias should be simple (bnc#768624)
+          p_alias = [p.display_name, p.name, "repo"].reject(&:empty?).first
```
Comment (imobachgs) [inline src/lib/installation/clients/inst_addon_update_sources.rb]: ... and an `#alias` method :)

**Pull request merged**